version: '3.1'

services:
  db:
    image: postgres:16
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U odoo"]
      interval: 10s
      timeout: 5s
      retries: 5
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=odoo
      - POSTGRES_PASSWORD=odbpwd
    volumes:
      - odoo-db-data:/var/lib/postgresql/data
    ports:
      - "5444:5432"
    restart: unless-stopped
    networks:
      - odoo-network

  web:
    image: odoo:17
    depends_on:
      db:
        condition: service_healthy
    ports:
      - "8070:8069"
    volumes:
      - ./config:/etc/odoo
      - ./addons:/mnt/extra-addons
      - odoo-web-data:/var/lib/odoo
    environment:
      - LANG=en_US.UTF-8
      - LANGUAGE=en_US:en
      - LC_ALL=en_US.UTF-8
      - HOST=db
      - POSTGRES_USER=odoo
      - POSTGRES_PASSWORD=odbpwd
      - POSTGRES_DB=postgres
    restart: unless-stopped
    networks:
      - odoo-network

volumes:
  odoo-db-data:
  odoo-web-data:

networks:
  odoo-network:
    driver: bridge